services/QuillDiagnostic/app/components/shared/uploadOptimalResponses.jsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react'
+import XLSX from 'xlsx'
+
+export default class UploadOptimalResponses extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { responses: [], }
+
+    this.handleChangeFile = this.handleChangeFile.bind(this)
+    this.submitResponses = this.submitResponses.bind(this)
+  }
+
+  handleChangeFile(file) {
+    const fileReader = new FileReader();
+    fileReader.onload = (e) => {
+      const data = new Uint8Array(e.target.result);
+      const workbook = XLSX.read(data, { type: 'array', });
+      const sheet = workbook.Sheets[workbook.SheetNames[0]]
```

Comment:
This seems obvious-ish to me, but it might be worth dropping some comments in here to explain why we're using which values.  Like here I presume we're just assuming single-sheet XLSX files, and thus need to find the first sheet to get content from.